### PR TITLE
Add some generic instrumentation to the ingestor

### PR DESCRIPTION
**Pull request by tomwilkie**
Reuses instrumentation from scope.  Clearly won't be suitable for upstreaming, but we can cross that bridge later...

### CODE REVIEW

File: frankenstein/notes.md [hunk 1]
```diff
@@ -3,17 +3,12 @@
 Design Doc: https://docs.google.com/document/d/1C7yhMnb1x2sfeoe45f4mnnKConvroWhJ8KQZwIHJOuw/edit#heading=h.f7lkb8wswewc
 
     GOOS=linux DOCKER_IMAGE_TAG=latest make build docker docker-frank
-    ./frankenstein/run.sh
```
Comment (jml) [inline frankenstein/notes.md:6]: How should people run it now?

**Comment by jml** [comment]
LGTM.

File: frankenstein/notes.md (hunk 1 above)
Comment (tomwilkie) [inline frankenstein/notes.md:6]: They should use weave cloud...

But seriously, this old method had bit rotted and was misleading.  Id rather have no instructions than instructions that don't work.

File: frankenstein/cmd/frankenstein/main.go [hunk 2]
```diff
@@ -205,9 +219,17 @@ func setupIngester(
 		log.Fatal(err)
 	}
 	prometheus.MustRegister(ingester)
-	http.Handle("/push", frankenstein.AppenderHandler(ingester))
-	http.Handle("/query", frankenstein.QueryHandler(ingester))
-	http.Handle("/label_values", frankenstein.LabelValuesHandler(ingester))
+
+	instr := middleware.Merge(
+		middleware.Logging,
```
Comment (juliusv) [inline frankenstein/cmd/frankenstein/main.go:224]: Do we really want to log every sample push?  How about only tracking those via metrics?

Also, I wonder if it's worth pulling in a 2700+ line dependency (gorilla and Weaveworks dep) just for a small bit of instrumentation.

**Pull request merged**